framework/src/bundle/BundleUtils.h

```diff
@@ -34,6 +35,15 @@ void* GetExecutableHandle();
 
 // returns the address of the symbol in library libHandle
 void* GetSymbol(void* libHandle, const char* symbol);
+
+template<typename T> void GetSymbol(std::function<T>& fptr,
+                                    void* libHandle,
+                                    const std::string& symbol)
+{
+    void* f = GetSymbol(libHandle, symbol.c_str());
+    fptr = reinterpret_cast<T*>(f);
```

Comment:
I remember that the previous implementation (using memcpy) was the only that compiled without warnings on all compilers / platforms. Is that still the case? Was there a particular issue why the implementation was changed to usingin `reinterpret_cast`?

Reply:
It's my understanding that that's the purpose of reinterpret_cast. Specifically, behavior 7) as described on the reference site: " Any pointer to function can be converted to a pointer to a different function type. Calling the function through a pointer to a different function type is undefined, but converting such pointer back to pointer to the original function type yields the pointer to the original function." Then the hook pointers are stored in a std::function object rather than simply a pointer to a function. This makes the code at the call site cleaner. I didn't notice any compiler warnings.

Reply:
I am aware of how reinterpret_cast is documented (at least nowadays). Still, there was a reason for using the memcpy trick. If you say that is all fine now, even better!

Reply:
Sorry if I sounded condescending. I didn't mean to! I looked up "reinterpret_cast" to make sure that *I* understood it correctly :-)

Reply:
Don't worry. However, even if the documentation says we can do this (on POSIX), it doesn't mean that the tool-chains we use fully comply with that. Anyway, if warnings are popping up from this later, we can fix it on demand.